kolibri/plugins/coach/assets/src/views/plan/CreateExamPage/SectionSidePanel.vue

```diff
@@ -0,0 +1,52 @@
+<template>
+
+  <SidePanelModal
+    v-if="$route.params.section_id"
+    ref="resourcePanel"
+    alignment="right"
+    :closeButtonIconType="closeIcon"
+    @closePanel="$router.back()"
+    @shouldFocusFirstEl="findFirstEl()"
+  >
+    <component :is="panel" :ref="$route.name" />
+  </SidePanelModal>
+
+</template>
+
+
+<script>
+
+  import SidePanelModal from 'kolibri-common/components/SidePanelModal';
+  import { PageNames } from '../../../constants';
+  import SectionEditor from './SectionEditor';
+  import ReplaceQuestions from './ReplaceQuestions';
+  import ResourceSelection from './ResourceSelection';
+
+  const pageNameComponentMap = {
+    [PageNames.QUIZ_SECTION_EDITOR]: SectionEditor,
+    [PageNames.QUIZ_REPLACE_QUESTIONS]: ReplaceQuestions,
+    [PageNames.QUIZ_SELECT_RESOURCES]: ResourceSelection,
+  };
+
+  export default {
+    name: 'SectionSidePanel',
+    components: { SidePanelModal, SectionEditor, ReplaceQuestions, ResourceSelection },
+    computed: {
+      panel() {
+        return pageNameComponentMap[this.$route.name];
+      },
+      closeIcon() {
+        return this.$route.name === PageNames.QUIZ_SELECT_RESOURCES ? 'back' : 'close';
+      },
+    },
+    methods: {
+      /**
+       * Calls the currently displayed ref's focusFirstEl method.
+       */
+      findFirstEl() {
+        this.$refs[this.$route.name].focusFirstEl();
```

Comment:
It looks like this seems to break the modal some times. The modal failed to show some times on click. Please see a screenshot below;

<img width="1914" alt="Screenshot 2023-08-31 at 20 42 44" src="https://github.com/learningequality/kolibri/assets/5203639/84b916c5-00b3-4ddc-9845-1d9e437b399b">